src/main/java/com/robertx22/mine_and_slash/capability/entity/EntityData.java

```diff
@@ -959,6 +960,14 @@ public void onSpellHitTarget(Entity spellEntity, LivingEntity target) {
 
     }
 
+    // Tracks LOSS of resources (spend, drains, damage, etc.)
+    private final ResourceTracker resourceTracker = new ResourceTracker();
+    public ResourceTracker getResourceTracker() { return resourceTracker; }
+
+    public final SpendThresholdRuntime spendRuntime = new SpendThresholdRuntime();
```

Comment:
To keep the same code style
```suggestion
    private final SpendThresholdRuntime spendRuntime = new SpendThresholdRuntime();
```